webpack.config.js

```diff
@@ -0,0 +1,179 @@
+const path = require("path");
+const HtmlWebPackPlugin = require("html-webpack-plugin");
+const ReactRefreshWebpackPlugin = require("@pmmmwh/react-refresh-webpack-plugin");
+const TerserPlugin = require("terser-webpack-plugin");
+const CopyPlugin = require("copy-webpack-plugin");
+const ForkTsCheckerWebpackPlugin = require("fork-ts-checker-webpack-plugin");
+const { createBabelConfig } = require('./babel.config.js');
+const { DefinePlugin, ProvidePlugin } = require("webpack");
+const JSONManifestPlugin = require("./jsonmanifest-plugin");
+
+const createStats = verbose => ({
+    assets: verbose,
+    builtAt: verbose,
+    cached: false,
+    children: false,
+    chunks: false,
+    colors: true,
+    entrypoints: true,
+    hash: false,
+    modules: false,
+    performance: false,
+    timings: verbose,
+    version: verbose,
+});
+
+module.exports = (env, argv) => ({
+    context: path.resolve(__dirname),
+    // Our application entry point.
+    entry: {
+        main: [
+            "./src",
+        ],
+    },
+    devtool: 'inline-source-map',
+    module: {
+        rules: [
+            {
+                test: /\.[jt]sx?$/,
+                exclude: /node_modules/,
+                use: [
+                    {
+                        loader: "babel-loader",
+                        options: createBabelConfig({ mode: argv.mode }),
+                    },
+                ],
+            },
+            {
+                test: /\.css$/,
+                exclude: /node_modules/,
+                use: ["style-loader", "css-loader"],
+                sideEffects: true,
+            },
+            {
+                test: /\.(scss)$/,
+                use: ["style-loader", "css-loader", "postcss-loader", "fast-sass-loader"],
+                sideEffects: true,
+            },
+            {
+                test: /\.(png|jpe?g|gif|svg)$/i,
+                loader: 'file-loader',
+            },
+        ],
+    },
+
+    resolve: {
+        extensions: ['.tsx', '.ts', '.jsx', '.js'],
+        // polyfill
+        fallback: {
+            http: require.resolve("stream-http"),
+            https: require.resolve("https-browserify"),
+            buffer: require.resolve("buffer"),
+            util: require.resolve("util"),
+            process: require.resolve("process/browser"),
+        },
+    },
+
+    output: {
+        path: path.resolve(__dirname, "./dist"), // cwp cant read this, it's dumb (even though this is the default as of wbpack5)
+        filename: `[name].bundle.js`,
```

Comment:
Keep the content hash on prod
```suggestion
        filename: `[name]${argv.mode === 'production' ? ".[contenthash]" : ""}.js`,
```